java-api-core/src/com/kinvey/java/store/requests/data/save/SaveListRequest.java

```diff
@@ -51,16 +53,42 @@ public SaveListRequest(ICache<T> cache, NetworkManager<T> networkManager, WriteP
 
     @Override
     public List<T> execute() throws IOException {
-        List<T> ret = new ArrayList<T>();
+        List<T> ret = new ArrayList<>();
         switch (writePolicy) {
             case FORCE_LOCAL:
                 ret = cache.save(objects);
                 syncManager.enqueueRequests(networkManager.getCollectionName(), networkManager, RequestMethod.SAVE, ret);
                 break;
             case LOCAL_THEN_NETWORK:
+                PushRequest<T> pushRequest = new PushRequest<>(networkManager.getCollectionName(), cache, networkManager,
+                        networkManager.getClient());
+                try {
+                    pushRequest.execute();
+                } catch (Throwable t) {
+                    t.printStackTrace();
+                    // silent fall, will be synced next time
+                }
+                List<T> notPushedObjects = new ArrayList<>();
+                IOException exception = null;
+                for (T object : objects) {
+                    try {
+                        ret.add(networkManager.saveBlocking(object).execute());
+                    } catch (IOException e) {
+                        syncManager.enqueueRequest(networkManager.getCollectionName(),
+                                networkManager, RequestMethod.SAVE, (String) object.get(Constants._ID));
+                        notPushedObjects.add(object);
+                        exception = e;
+                    }
+                }
+                cache.save(ret);
```

Comment:
@yuliya-guseva In the LOCAL_THEN_NETWORK case, is the local cache save happening after the network request? If so, it may make more sense to first save locally to cache before attempting the network operation.